app/src/main/java/com/team7/cmput301/android/theirisproject/activity/SplashActivity.java

```diff
@@ -4,20 +4,110 @@
+        IrisProjectApplication.setApplicationContext(getApplicationContext());
+        controller = (LoginController) createController(getIntent());
+        IrisProjectApplication.initBulkUpdater();
+
+        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(this);
+        String username = sharedPref.getString(getString(R.string.shared_pref_username_key), "");
+
+        Log.d(TAG, "username is " + username);
+
+        // Redirect to login/register activity because this user isn't valid
+        if (username == null || username.isEmpty()) {
+            finish();
+            startActivity(new Intent(this, LoginActivity.class));
+        } else {
+            // Otherwise if username exists in DB, login with that user and wait until we load their data
+            controller.loginUser(username, new Callback<Boolean>() {
+                @Override
+                public void onComplete(Boolean success) {
+                    // Start ProblemList or PatientList activity if login is successful,
+                    // otherwise redirect to login/register activity
+                    if (success) {
+                        Toast.makeText(SplashActivity.this, getString(R.string.login_success), Toast.LENGTH_LONG).show();
+                        buildUserSession();
+                        // Build user session will start the correct User activity
+                    } else {
+                        finish();
+                        startActivity(new Intent(SplashActivity.this, LoginActivity.class));
+                    }
+                }
+            });
+        }
+
+    }
+
+    /**
+     * Push the update queue backups to online, get all the logged in User's associated data,
+     * then start a specified activity for them.
+     */
+    private void buildUserSession() {
+
+        Callback callbackToStart = new Callback() {
+            @Override
+            public void onComplete(Object res) {
+                Class activity = controller.getStartingActivity();
+                // Put finish here so that the splash screen stays until we're done loading
+                finish();
+                startUserActivity(activity);
+            }
+        };
+
+        Callback<Boolean> callbackToFetch = new Callback<Boolean>() {
+            @Override
+            public void onComplete(Boolean res) {
+                controller.fetchAllUserData(callbackToStart);
+            }
+        };
+
+        IrisProjectApplication.flushUpdateQueueBackups(callbackToFetch);
+
+    }
+
+    /**
+     * startUserActivity takes in the targetActivity to goto if
+     * login is successful.
+     *
+     * @param targetActivity: an activity class for the intent
+     * */
+    private void startUserActivity(Class<?> targetActivity) {
+        Intent intent = new Intent(SplashActivity.this, targetActivity);
+        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);  // prevents multiple occurrences
```

Comment:
I think this isn't needed, since there is no user input in this activity

Reply:
This activity uses it in buildUserSession